lib/ocpp/v201/smart_charging.cpp

```diff
@@ -193,13 +194,22 @@ SmartChargingHandler::validate_profile_schedules(ChargingProfile& profile,
 }
 
 SetChargingProfileResponse SmartChargingHandler::add_profile(int32_t evse_id, ChargingProfile& profile) {
-    if (STATION_WIDE_ID == evse_id) {
-        station_wide_charging_profiles.push_back(profile);
-    } else {
-        charging_profiles[evse_id].push_back(profile);
-    }
     SetChargingProfileResponse response;
     response.status = ChargingProfileStatusEnum::Accepted;
+
+    auto& profile_storage = STATION_WIDE_ID == evse_id ? station_wide_charging_profiles : charging_profiles[evse_id];
+    auto pos = std::distance(profile_storage.begin(), std::find_if(profile_storage.begin(), profile_storage.end(),
+                                                                   [&profile](const ChargingProfile existing_profile) {
+                                                                       return profile.id == existing_profile.id;
+                                                                   }));
+
+    if (pos >= profile_storage.size() ||
```

Comment:
Is there a nicely named function we can have here that describes what the first part of this if statement is doing at a higher level? 

profileIDsDoNotMatch() maybe?

Reply:
Updated per our discussion on Teams :)